app-experimental.tex

```diff
@@ -2395,3 +2395,219 @@ \section{Distributed Capabilities For Peripherals And Accelerators} % <<<
 % >>>
 
 % vim: foldmethod=marker:foldmarker=<<<,>>>
+
+% >>>
+% >>>
+\section{Compartment ID Sealing} % <<<
+
+\subsection{Motivation}
+
+Compartment identification is essential. Code and data associated with a compartment must be somehow identified and validated when they are being accessed.
+The most natural way seems to be to give them identification numbers.
+This has traditionally been done in software, e.g., processes are assigned process IDs (PIDs) in operating systems. In some cases, a software ID can correspond to a hardware ID, e.g., PIDs can correspond to ASIDs.
+However, that is not necessarily the case.
+In this proposal, we suggest an architectural ID for compartments.
+We call these numbers compartment IDs (CIDs). That means that one CID is mapped to one compartment at a time.
+An architecturally defined CID needs to be implemented by hardware.
+
+\subsection{Storage}
+
+The first question to tackle is where to store a CID. We envision multiple options for that.
+First, one can use a dedicated register for compartment identification as done by the Morello architecture.
+Second, one can import CIDs into the capability format, which is what we propose in this document.
+Having this solution leads to an atomic change of CID and code capability -- a property we envision to be useful for secure compartmentalisation.
+
+The CID will substitute the otype bits (18 bits) in the 128-bit capability format.
+The otype bits are currently not well-used. The otype field indicates whether a capability is sealed and if so which “type” it has.
+There exists currently two fixed values that are used: one for unsealed capabilities and one for sealed entry (sentry) capabilities.
+There are 14 more reserved fixed values, which are not currently used. All other values are used as values for sealed capabilities.
+
+However, there is also the possibility to combine otypes and CIDs.
+One potential approach is to subdivide the 18 otype bits such that there is space for a few otypes and the remainder of the bits is dedicated to CIDs.
+This would allow sentries and CID sealing to be combined.
+
+In the following text, we propose to completely substitute otypes.
+All values but the zero CID (CID==0) are valid IDs.
+Furthermore, every CID sealed code capability already is a sealed entry capability.
+Please note that every capability within a compartment can be manipulated by the compartment itself.
+This also includes CID sealed code capabilities.
+In the CHERI world up to now, sentries cannot be manipulated.
```

Comment:
No proposal with an orthogonal sentry bit?

Reply:
Good point, that should be part of the idea space - not sure where I got lost. I will put that in.